adding containers concept page (#152)
Basis-Theory/basistheory-integration-guides: collections/concepts/what-are-containers.md

---
layout: post
title:  Token Containers
categories: concepts
permalink: /concepts/what-are-token-containers/
nav_order: 3
has_children: true
has_toc: false
image:
path: https://cdn.basistheory.com/images/seo/guides-opengraph.png
width: 1200
height: 630
---

# What are Token Containers?

Token Containers are hierarchical grouping constructs that allow you to logically segment tokens within your tenant. With

> this grouping in place, you are then able to control access to your tokens based on [Access Rules](https://developers.basistheory.com/concepts/access-controls/) 

Reviewer: ```suggestion
this grouping in place, you are able to control access to your tokens based on [Access Rules](https://developers.basistheory.com/concepts/access-controls/) 
```

> tied to applications that specify which containers an Application has access to. 

Author: i don't really like to capitalize the plural form of the entity, if it was Token or Application it makes sense but for plural I think we should use lowercase

> tied to applications that specify which containers an Application has access to. 

Reviewer: what's your reasoning for that? I still see them as resources that we capitalize

> Token Containers are hierarchical grouping constructs that allow you to logically segment tokens within your tenant. With
> this grouping in place, you are able to control access to your tokens based on [Access Rules](https://developers.basistheory.com/concepts/access-controls/) 
> tied to applications that specify which containers an Application has access to. 

Reviewer 2: The wording here is confusing and overly complex.

```suggestion
Tokens can be logically grouped into Containers to segment access to tokens within a Tenant. Fine-grained access control policies can be defined by attaching one or more [Access Rules](https://developers.basistheory.com/concepts/access-controls/) to an Application. Each Access Rule is scoped to a specific Container of tokens, allowing you to grant an Application permission to interact with only a subset of tokens within your Tenant.
```

A container is represented by a hierarchical path, which is conceptually similar to directories in a UNIX filesystem.
Container names must start and end with a `/`, and the root container is denoted by `/`. Container names may include any 
alphanumeric characters, `-`, or `_`, and can contain an arbitrary number of nested sub-containers.

You have complete control to customize your container hierarchy to meet your unique data governance requirements. By default,

> tokens will be assigned to containers based off their data classification and impact levels, using the format 

Reviewer: ```suggestion
Tokens will be assigned to containers based off their data classification and impact levels, using the format 
```

> tokens will be assigned to containers based off of their data classification and impact levels, using the format 

Reviewer: ```suggestion
tokens will be assigned to containers based off their data classification and impact levels, using the format 
```

`/<classification>/<impact_level>/`, e.g. `/pci/high/` or `/general/low/`.

---

## Common Use Cases

### Segmenting by data classification

> If you need to segment tokens by data classification, you could organize your tokens into the following containers:

Reviewer: ```suggestion
If you need to segment Tokens by data classification, you could organize your Tokens into the following containers:
```

- `/pci/`
- `/bank/`
- `/pii/`
- `/general/`
- `/my-custom-classification/`

---

### Segmenting by customer

> If you need to segment tokens by customer, you could organize your tokens into the following containers:

Reviewer: ```suggestion
If you need to segment Tokens by customer, you could organize your tokens into the following containers:
```
